### PR TITLE
Adding the restarting for multisegment wells

### PR DESCRIPTION
**Pull request by GitPaean**

### CODE REVIEW

File: opm/autodiff/BlackoilModelParametersEbos.hpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ SET_SCALAR_PROP(FlowModelParameters, ToleranceCnvRelaxed, 1e9);
 SET_SCALAR_PROP(FlowModelParameters, ToleranceWells, 1e-4);
 SET_SCALAR_PROP(FlowModelParameters, ToleranceWellControl, 1e-7);
 SET_INT_PROP(FlowModelParameters, MaxWelleqIter, 15);
-SET_BOOL_PROP(FlowModelParameters, UseMultisegmentWell, false);
+SET_BOOL_PROP(FlowModelParameters, UseMultisegmentWell, true);
```
Comment (GitPaean) [inline opm/autodiff/BlackoilModelParametersEbos.hpp]: oh, I need to change this one to be false.

**Comment by GitPaean** [comment]
I will clean up a little bit.

**Comment by GitPaean** [comment]
I am done updating. Please review. I think it can go in together along with https://github.com/OPM/opm-simulators/pull/1592 .

File: opm/autodiff/BlackoilWellModel_impl.hpp [hunk 2]
```diff
@@ -1724,6 +1723,85 @@ namespace Opm {
                 }
                 ++local_comp_index;
             }
+
+            if (handle_ms_well && !well.segments.empty()) {
+                // we need the well_ecl_ information
+                const std::string& well_name = wm.first;
+                const Well* well_ecl = getWellEcl(well_name);
+                assert(well_ecl);
```
Comment (GitPaean) [inline opm/autodiff/BlackoilWellModel_impl.hpp:1731]: `assert` here is kind of temporary approach. I have not been able to investigate the exception system introduced in https://github.com/OPM/opm-simulators/pull/1730. If it is needed to follow the exception system, I will fix the `assert` here and the one below.

**Comment by bska** [comment]
Okay, thanks.  Do you think you could squash your commits down to one and provide a more detailed description?

**Comment by GitPaean** [comment]
I squashed the commits and write a little more description.

**Comment by bska** [comment]
> I squashed the commits and write a little more description.

Thanks.  I'll merge this into my branch to make it a part of OPM/opm-simulators#1592.

**Pull request merged**